R/zoom.R

```diff
@@ -0,0 +1,22 @@
+cdm_zoom_to_tbl <- function(dm, table) {
+  # for now only one table can be zoomed on
+  zoom <- as_string(ensym(table))
+  check_correct_input(dm, zoom)
+  zoom_tbl <- tbl(dm, zoom)
+
+  new_dm2(
+    zoom = tibble(table = zoom, zoom = list(zoom_tbl)),
+    base_dm = dm
+  )
+}
+
+is_zoomed <- function(dm) {
+  !all(cdm_get_zoomed_tbl(dm) %>% pull(zoom) %>% map_lgl(is_null))
```

Comment:
Can you implement using `cdm_get_zoomed_tbl()` ?

Reply:
I would think I kind of did. The name for the function `cdm_get_zoomed_tbl()` is maybe a bit misleading: it does not return only the tibble itself, but a tibble with two columns and one row: the original table name and a list with the zoomed table.

I see your point and I think we need two functions:
1. the function as it is currently implemented (it's very similar to `cdm_get_filter()`) and it should be exported eventually
2. an internal function (e.g. `get_zoomed_tbl()`), that just returns the zoomed table if one exists, and `NULL` otherwise. <- this one I can then use for `is_zoomed()`

Reply:
Sorry, missed that. How about `nrow(...) > 0` ?

Reply:
I was now thinking of introducing a subclass `zoomed_dm`, similar to `grouped_df`. Would that be an option too? Then the printing could work even more similarly to the printing of tibbles. Would that be ok with you?

Reply:
A subclass seems "cleaner", but let's implement without a subclass first and later think where it would help us.

Reply:
I think one important argument for introducing the subclass is, that the {dplyr} verbs would work with the subclass "zoomed_dm", but not with "dm".

Reply:
FYI: It seemed just so much more straightforward to implement it using the subclass that I went for it. Also, as mentioned in my last comment in this thread, I really do believe it makes a lot of sense to have this subclass.